test/safe-stringify.spec.ts

```diff
@@ -23,6 +23,14 @@ test.group('Stringify', () => {
     assert.deepEqual(safeStringify(a), '{"b":2}')
   })
 
+  test('stringify object with set', (assert) => {
+    const a: any = {
+      b: 2,
+      a: new Set([1, 2, 3]),
```

Comment:
Will it give desired outcome when you have nested sets?

```ts
new Set([1, new Set([2]), 3])
```

Reply:
No, that code would output the following:
```ts
[ 1, Set(1), 3 ]
```

Since `Set` can be quite versatile, we cannot handle all use cases. I am not sure it will be useful outside of first-level serialization.

Reply:
That's not what I experience. I just tried this with your PR and it works:

```ts
  test('stringify nested sets', (assert) => {
    const a: any = new Set([1, new Set([{ hello: new Set(['world']) }]), 'test'])
    assert.deepEqual(safeStringify(a), '[1,[{"hello":["world"]}],"test"]')
  })
```

Reply:
Oh great! I only tried in my console with `Array.from()`, which is not the complete implementation of what `safeStringify` does.

If you want, I can add some test cases of complex `Set`.